apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/SNFClaimTransformerV2.java

```diff
@@ -314,21 +312,18 @@ private ExplanationOfBenefit transformClaim(SNFClaim claimGroup) {
     // FST_DGNS_E_VRSN_CD       => diagnosis.diagnosisCodeableConcept
     // ICD_DGNS_E_CD(1-12)      => diagnosis.diagnosisCodeableConcept
     // ICD_DGNS_E_VRSN_CD(1-12) => diagnosis.diagnosisCodeableConcept
-    for (Diagnosis diagnosis : DiagnosisUtilV2.extractDiagnoses(claimGroup)) {
-      DiagnosisUtilV2.addDiagnosisCode(eob, diagnosis, ClaimTypeV2.SNF);
-    }
+    DiagnosisUtilV2.extractDiagnoses(
+            claimGroup.getDiagnosisCodes(), claimGroup.getDiagnosisCodeVersions(), Optional.empty())
+        .stream()
+        .forEach(diagnosis -> DiagnosisUtilV2.addDiagnosisCode(eob, diagnosis, ClaimTypeV2.SNF));
 
     // Handle Procedures
-    // ICD_PRCDR_CD(1-25)        => ExplanationOfBenefit.procedure.procedureCodableConcept
-    // ICD_PRCDR_VRSN_CD(1-25)   => ExplanationOfBenefit.procedure.procedureCodableConcept
-    // PRCDR_DT(1-25)            => ExplanationOfBenefit.procedure.date
-    final int FIRST_PROCEDURE = 1;
-    final int LAST_PROCEDURE = 25;
-
-    IntStream.range(FIRST_PROCEDURE, LAST_PROCEDURE + 1)
-        .mapToObj(i -> TransformerUtilsV2.extractCCWProcedure(i, claimGroup))
-        .filter(p -> p.isPresent())
-        .forEach(p -> TransformerUtilsV2.addProcedureCode(eob, p.get()));
+    TransformerUtilsV2.extractCCWProcedures(
+            claimGroup.getProcedureCodes(),
+            claimGroup.getProcedureCodeVersions(),
+            claimGroup.getProcedureDates())
+        .stream()
+        .forEach(p -> TransformerUtilsV2.addProcedureCode(eob, p));
```

Comment:
I like the new code here, cleaner than the way it was done before. (I see it's still done with the counter in the underlying code, so its not totally gone, but at least its abstracted)